### PR TITLE
Cut Minischedule.Block

### PR DESCRIPTION
**Pull request by phildarnowsky**
Card: https://app.asana.com/0/1200180014510248/1200611799543200

### CODE REVIEW

File: test/schedule/minischedule/load_test.exs [hunk 1]
```diff
@@ -5,39 +5,32 @@ defmodule Schedule.Minischedule.LoadTest do
...
   alias Schedule.AsDirected
-  alias Schedule.Minischedule.Block
   alias Schedule.Break
   alias Schedule.Minischedule.Load
   alias Schedule.Piece
   alias Schedule.Minischedule.Run
 
-  describe "from_hastus/3" do
-    test "trips become pieces in run and block" do
+  describe "runs_from_hastus/3" do
```
Comment (lemald) [inline test/schedule/minischedule/load_test.exs:13]: Should probably also add a test for `blocks_from_runs` (unless that simply goes away in a future commit).

File: lib/realtime/vehicle.ex [hunk 2]
```diff
@@ -331,8 +331,9 @@ defmodule Realtime.Vehicle do
 
   def route_status(stop_id, trip, block) do
     if stop_id == List.first(trip.stop_times).stop_id do
+      first_trip_from_block = block && block |> Block.revenue_trips() |> List.first()
       # hasn't started trip yet
-      if block != nil && trip.id == List.first(block.trips).id do
+      if block != nil && trip.id == first_trip_from_block.id do
```
Comment (lemald) [inline lib/realtime/vehicle.ex:336]: I think you can get rid of the `block != nil` given the logic behind `final_trip_from_block`, right?

**Comment by lemald** [comment]
I've reviewed the first few commits and will finish my review up tomorrow morning. Really cool to see this PR open, though, this is arguably the meat of the whole refactor!

File: test/concentrate/consumer/stop_time_updates_test.exs [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-    id: "S28-2",
-    service_id: "service",
-    start_time: 0,
-    end_time: 0,
-    trips: [@trip]
-  }
+  @block build(:block,
+           id: "S28-2",
+           end_time: 0
+           # trips: [@trip]
```
Comment (phildarnowsky) [inline test/concentrate/consumer/stop_time_updates_test.exs:20]: note to self: cut

File: lib/schedule/piece.ex [hunk 4]
```diff
@@ -58,6 +58,25 @@ defmodule Schedule.Piece do
...
+  #
+  # These trips will have a nil service_id. So do all non-revenue trips, but
+  # since a piece should always have at least one revenue trip, if all trips
+  # in a piece have a nil service_id, we can assume that all those trips are
+  # from a non-current rating.
+  @spec from_non_current_rating?(t()) :: boolean()
+  def from_non_current_rating?(piece) do
+    piece.trips
+    |> Enum.reject(&match?(%AsDirected{}, &1))
+    |> Enum.all?(&(&1.service_id == nil))
```
Comment (lemald) [inline lib/schedule/piece.ex:77]: Small nitpick - could do `is_nil(&1.service_id)`.

File: lib/schedule/trip.ex [hunk 5]
```diff
@@ -173,4 +173,9 @@ defmodule Schedule.Trip do
   end
 
   def set_pretty_names(trip, _), do: trip
+
+  @spec is_revenue_trip?(t()) :: boolean()
+  def is_revenue_trip?(trip) do
+    (trip.service_id && trip.route_id) != nil
```
Comment (lemald) [inline lib/schedule/trip.ex:179]: I think this would be a bit clearer if written as `!is_nil(trip.service_id) && !is_nil(trip.route_id)`.

**Comment by lemald** [comment]
@phildarnowsky okay, whew, I'm done with my first pass of reviewing this. I have a few minor comments but otherwise this is really, really cool!

File: test/schedule/minischedule/load_test.exs (hunk 1 above)
Comment (phildarnowsky) [reply test/schedule/minischedule/load_test.exs:13]: it does disappear

File: lib/realtime/vehicle.ex (hunk 2 above)
Comment (phildarnowsky) [reply lib/realtime/vehicle.ex:336]: that should really be `if first_trip_from_block && trip.id == first_trip_from_block.id`

**Comment by phildarnowsky** [comment]
@lemald have responded to all your feedback, I'm also going to push this (slightly) updated version to dev-blue to bake over the weekend

**Review by lemald — APPROVED** [review]
Looks good! 🎉

**Comment by phildarnowsky** [comment]
@lemald FYI going to close this PR for neatness, on the grounds that it's been superseded by #1240

**Pull request closed without merging**